src/audio/kpb.c

```diff
@@ -810,6 +810,36 @@ static int kpb_prepare(struct comp_dev *dev)
 			break;
 		}
 	}
+#else
+	/* Update number of sel_sink channels.
+	 * If OBS is not equal to IBS it means that KPB will work in micselector mode.
+	 */
+	if (kpb->ipc4_cfg.base_cfg.ibs != kpb->ipc4_cfg.base_cfg.obs) {
+		struct list_item *sink_list;
+		const uint32_t byte_align = 1;
+		const uint32_t frame_align_req = 1;
+		uint32_t sink_id;
+
+		if (!list_is_empty(&dev->bsink_list)) {
```

Comment:
this isn't needed, please, remove. If the list is empty, the loop won't execute, so the effect is exactly the same

Reply:
Done